.github/workflows/sorted-pr-checks.yml

```
@@ -0,0 +1,32 @@
+name: Comment with sorted PR checks
+
+on:
+  workflow_dispatch:
+    inputs:
+      pull_number:
+        description: 'Pull request number'
+        required: true
+  workflow_run:
+    workflows:
+      - Build
+      - Check
+      - CodeQL
+      - Test
+    types:
+      - requested
+      - completed
+
+permissions:
+  pull-requests: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.inputs.pull_number || github.event.workflow_run.pull_requests[0].number || 'unknown' }}
+  cancel-in-progress: true
+
+jobs:
+  comment:
+    if: github.event.inputs.pull_number || github.event.workflow_run.pull_requests[0]
+    uses: ipdxco/sorted-pr-checks/.github/workflows/comment.yml@v1
+    with:
+      pull_number: ${{ github.event.inputs.pull_number || github.event.workflow_run.pull_requests[0].number }}
+      template: grouped_by_result
```

This is how we can switch to listing unsuccessful checks only. At the moment, that setting doesn't group the failures, though. If you want us to add that, just let me know :)

```suggestion
      template: unsuccessful_only
```

---

@galargh I think the general _vibe_ of the thread here is that we should do `unsuccessful_only` but it's good to know that `grouped_by_result` is a possibility if we want it. I assume that this means the sticky will only appear if there's unsuccessful and otherwise there'll be no noise?

---

It will always appear, even if all the checks pass. The comment appears as early as the workflows start to inform the user they are in progress. Once they're all finished, the action changes the content to match the workflows' outcomes. It's easier to implement that way and I believe it creates better expectations for the end user as they can always check the results in the comment.